File: java/food-ordering/app/restate-app/src/main/java/dev/restate/sdk/examples/types/OrderStatus.java
package dev.restate.sdk.examples.types;

class OrderStatus {
  private Status status;
  private long eta;

  public OrderStatus(Status status, long eta) {
    this.status = status;
    this.eta = eta;
  }

  public Status getStatus() {
    return status;
  }

  public long getEta() {
    return eta;
  }

  public OrderStatus setStatus(Status status) {
    this.status = status;
    return this;
  }

  public OrderStatus setEta(long eta) {
    this.eta = eta;
    return this;
  }
}

File: python/end-to-end-applications/food-ordering/app/ordering/order_status.py
# Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
#
# This file is part of the Restate examples,
# which is released under the MIT license.
#
# You can find a copy of the license in the file LICENSE
# in the root directory of this repository or package or at
# https://github.com/restatedev/examples/

from restate import VirtualObject, ObjectContext
import order_workflow as order_workflow

order_status = VirtualObject("order-status")


@order_status.handler()
async def get(ctx: ObjectContext):
    eta = await ctx.get("eta") or None
    status = (
        await ctx.workflow_call(order_workflow.get_status, ctx.key(), arg=None) or None
    )
    return {"eta": eta, "status": status}


@order_status.handler()
async def set_eta(ctx: ObjectContext, eta: int):
    ctx.set("eta", eta)


@order_status.handler()
async def event_handler(ctx: ObjectContext, eta: int):
    ctx.set("eta", eta)
